### PR TITLE
Add API to get the language in which a symbol is defined and the symbol provider for a source file

### PR DESCRIPTION
**Pull request by ahoppen**
This will be used by sourcekit-lsp’s cross-file rename.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/sourcekit-lsp/pull/1038

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/sourcekit-lsp/pull/1038

@swift-ci Please test Windows

**Comment by ahoppen** [comment]
https://github.com/apple/sourcekit-lsp/pull/1038

@swift-ci Please test Windows

File: Sources/IndexStoreDB/IndexStoreDB.swift [hunk 1]
```diff
@@ -275,6 +285,41 @@ public final class IndexStoreDB {
...
+      let mainFilePath = String(cString: indexstoredb_unit_info_main_file_path(unit))
+      if providerKind == .swift && mainFilePath != sourceFilePath {
+        // We have a unit that is "included" from Swift. This happens for header
+        // files that Swift files depend on. But Swift doesn't have includes and
+        // we shouldn't infer the header file's language to Swift based on this unit.
+        // Ignore it.
+        return true
+      }
+
+      if result == nil {
```
Comment (bnbarham) [inline Sources/IndexStoreDB/IndexStoreDB.swift:311]: Is this worth it? Would be a pretty weird state to be in, may as well just return the first.

**Review by bnbarham — APPROVED** [review]

File: include/IndexStoreDB/Index/StoreUnitInfo.h [hunk 2]
```diff
@@ -26,16 +27,18 @@ struct StoreUnitInfo {
...
   bool HasTestSymbols = false;
   llvm::sys::TimePoint<> ModTime;
+  Optional<SymbolProviderKind> SymProviderKind;
 
   StoreUnitInfo() = default;
   StoreUnitInfo(std::string unitName, CanonicalFilePath mainFilePath,
                 StringRef outFileIdentifier, bool hasTestSymbols,
-                llvm::sys::TimePoint<> modTime)
+                llvm::sys::TimePoint<> modTime, Optional
+                <SymbolProviderKind> SymProviderKind)
```
[the remark this answers is not in the document]
Comment (ahoppen) [reply include/IndexStoreDB/Index/StoreUnitInfo.h:36]: No, that was me messing up the formatting

File: Sources/IndexStoreDB/IndexStoreDB.swift (hunk 1 above)
Comment (ahoppen) [reply Sources/IndexStoreDB/IndexStoreDB.swift:311]: I don’t have strong opinions either way. `indexstoredb_index_units_containing_file` also shouldn’t return many files so not sure if exiting early really makes a difference.

**Pull request merged**